dna/models.py

```diff
@@ -814,6 +815,93 @@ def predict_regression(self, data, *, verbose=False):
 
         return predictions
 
+class LinearRegressionBaseline(RegressionModelBase):
+    def __init__(self, seed=0):
+        RegressionModelBase.__init__(self, seed=seed)
+        self.regressor = linear_model.LinearRegression()
+        self.fitted = False
+        self.pipeline_key = 'pipeline'
+        self.steps_key = 'steps'
+        self.prim_name_key = 'name'
+
+    def fit(self, data, *, validation_data=None, output_dir=None, verbose=False):
+        self.fitted = True
+        self.one_hot_primitives_map = self._one_hot_encode_mapping(data + validation_data)
+        X_data, y = self.prepare_data_for_regression(data)
+        self.regressor.fit(X_data, y)
+
+    def predict_regression(self, data, *, verbose=False):
+        if self.fitted is False:
+            raise ModelNotFitError('LinearRegressionBaseline not fit')
+            
+        X_data, y = self.prepare_data_for_regression(data) 
+        predictions = self.regressor.predict(X_data)
+        return predictions
+    
+    def prepare_data_for_regression(self, data):
+        full_data = pd.DataFrame(data)
+        y = full_data["test_f1_macro"]
+
+        # this takes a column of lists of ints and expands it out into a dataframe of ints
+        metafeature_df_raw = pd.DataFrame(full_data.metafeatures.values.tolist()).reset_index(drop=True)
+        metafeature_df = pd.DataFrame(np.nan_to_num(metafeature_df_raw.values))
+        assert np.isnan(metafeature_df.values).sum() == 0, "Was not able to impute the metafeatures: nans exist"
+        assert np.isinf(metafeature_df.values).sum() == 0, "Was not able to impute the metafeatures: infs exist"
```

Comment:
nans and infs should be handled in `preprocess_data` in `data.py`. If you are observing any nans or infs, let's fix the `preprocess_data` method.

Reply:
There will be infs and nans based on other categories -> one hot encoding primitives, etc.  I still feel like it's valid to check these methods before we get errors in later code.  But you're right, theoretically they should be fine (and probably are). I can take out the lines that impute so no one misunderstands.

Reply:
> There could be infs and nans based on other categories -> one hot encoding primitives, etc.

This is related to the above comment. There should not be any infs or nans in one-hot encoding.

The asserts are probably fine, but yes let's remove the imputing from here for sure.

Reply:
Sounds good, I'll take em out.